3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -7,20 +7,20 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
 // TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
```

Comment:
```suggestion
        reject(new Error("You didn't pass in a first name!"));
        return;
```

In this exercise you should return immediately after rejecting the promise. Otherwise `resolve()` will be called also. Although this has no effect because the outcome of a settled promise cannot be changed, returning here is better than relying on the promise intrinsic behaviour.

Reply:
Hey @remarcmij, thank you very much for your nice feedback.
Okay, I understand now why, but should I return here because I did not use `else`? or Is it better to `return` always after reject or resolve?

Reply:
Using `else` here would be fine too.